### ISSUE TITLE
accuracy and loss calculation in cats_and_dogs.ipynb

### ISSUE THREAD

**Issue by nsriniva03** [lucidrains/vit-pytorch#160]
Hi, 

Could you recheck the snippet of code the is used to calculate accuracy in the training loop and validation loop?

------------------------
Current Implementation
------------------------

    epoch_loss = 0
    epoch_accuracy = 0

    for data, label in tqdm(train_loader):
        ......
       .......
        loss = criterion(output, label)
       ......
       .......
        acc = (output.argmax(dim=1) == label).float().mean()
        epoch_accuracy += acc / len(train_loader)
        epoch_loss += loss / len(train_loader)

----------------------------------

I think the epoch_accuracy and epoch loss should be computed as follows:

    epoch_loss = 0
    epoch_accuracy = 0

    for data, label in tqdm(train_loader):
        ......
       .......
        loss = criterion(output, label)
       ......
       .......
        acc = (output.argmax(dim=1) == label).float().mean()
        epoch_accuracy += acc
        loss += loss
 epoch_accuracy += acc / len(train_loader)
epoch_loss += loss / len(train_loader)
